zaps/0002-encrypt-push-notifications.md

```diff
@@ -174,6 +187,7 @@ bouncer (or Google/Apple's servers, for that matter).
   - This allows the client to determine if the `push_account_id` it has
     locally for the account is missing on the `push_account_ids` map.
     If missing, it knows it needs to reregister the push device.
+    It helps to show an error to the user in case of liveness error.
```

Comment:
I don't understand the last sentence here. What's a "liveness error" in this context? Is this saying that something helps for showing an error, or that showing an error is helpful?

Reply:
It's related to the error bouncer raises during registration:
>    **On error**:
>    - If it's a liveness error, don't retry and forget the `push_account_id`.
     Pass the error to the server admin. The user is notified about this
     error on the client using the `push_account_ids` map provided in the
     [register-queue](https://zulip.com/api/register-queue) response. (Refer
     to the [Miscellaneous](#miscellaneous) section for more details.)

Reply:
Hmm, I still don't understand what this is saying. Can you clarify this part?

> Is this saying that something helps for showing an error, or that showing an error is helpful?

And if the former: what is the thing that helps for showing an error?

Reply:
This is addressing https://github.com/zulip/zulip-architecture/pull/2#discussion_r2050112579

> Is this saying that something helps for showing an error

Yes.

>  if the former: what is the thing that helps for showing an error?

`push_account_ids` map (which has `push_account_id` and `status`) will help the client.

Reply:
Great, so this sentence could say:
> This data helps the client to show an error to the user in case of a liveness error.

I understand now what the sentence is saying. Next is the how and the why: how does the client use this data for that purpose, and why is that something we want to do? Why does the client care that there was specifically a liveness error?

For giving the user feedback that notifications aren't successfully set up, I think we probably want just a timeout: the client started trying to tell the server about its token more than, like, a few minutes ago, and the status is still that it hasn't succeeded yet.

Reply:
PUSH_REGISTRATION_LIVENESS_TIMEOUT is 24 hours. 

So, are you suggesting:
* Storing the timestamp in client when the registration request was made and server responded with 200.
* Then, in future, when the user reopens the map and the `push_account_id` is missing in the map + 24 hours passed; we can show the error and retry.

Reply:
No, I meant what I said.

If the user installed the Zulip app and logged into Zulip 5 minutes ago, and they're not getting notifications when people DM them, they're going to be wondering what's up because "the mobile app" clearly isn't working. So the app should acknowledge that.

At that point the server will still be periodically attempting to retry the registration with the bouncer (because PUSH_REGISTRATION_LIVENESS_TIMEOUT hasn't expired). In other words it will be still attempting to recover from the outage. But that doesn't mean there isn't an outage.

Reply:
Okay, simply client side handling.

Reply:
Updated in the 7th commit.